verilog/analysis/checkers/always_ff_non_blocking_rule.cc

```diff
@@ -60,6 +60,9 @@ void AlwaysFFNonBlockingRule::HandleSymbol(const verible::Symbol &symbol,
   if (always_ff_matcher_.Matches(symbol, &manager)) {
     for (const auto &match :
          SearchSyntaxTree(symbol, NodekAssignmentStatement())) {
+
+      if (match.context.IsInside(NodeEnum::kLoopHeader)) continue ;
```

Comment:
comment here: // this is intended to ignore assignments in for loop step statements like i=i+1

Reply:
Added comment. Thanks